DOC-847: Updated docs for table. (#2025)
tinymce/tinymce-docs: plugins/opensource/table.md

});
```

### `table_border_widths`

{{site.requires_5_9v}}

This option is used to specify a list of pre-defined cell border widths for quick access on the `tablecellborderwidth` toolbar button or menu item. This option accepts any valid CSS numeric value. [MDN web docs - Numeric data types](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Values_and_Units#numeric_data_types)

**Type:** `Array`

**Default Value:** 
```
[
  {title: '1px', value: '1px'},
  {title: '2px', value: '2px'},
  {title: '3px', value: '3px'},
  {title: '4px', value: '4px'},
  {title: '5px', value: '5px'}
]
```

#### Example: Using `table_border_widths`

```js
tinymce.init({
  selector: 'textarea',  // change this value according to your HTML
  plugins: 'table',
  menubar: 'table',
  toolbar: 'table',
  table_border_widths: [
    {title: '1 pixel', value: '1px'},
    {title: '1 Em', value: '1em'},

>     {title: '1 Vh', value: '1vh'},

Reviewer: Do we actually support this/have you tested this works as expected? IMO we should not be using an example like this and should just stick to the basics using pixels (e.g. use 1px, 3px, 5px or different labels such as small, medium, large instead).

- [ ] Give better/more realistic examples instead of just arbitrary data
- [ ] Ensure this is correct and that we do support all CSS units

Reviewer 2: https://github.com/tinymce/tinymce-docs/pull/2025#discussion_r674392533

It was my suggestion to use the example as an opportunity to show other CSS units (The documentation said any CSS unit is valid) but I agree we should use units that people would realistically use

Author: So do we want to keep to one thing, like px, or do want to mix? Since there seems to be some different opinions on it.

Reviewer: We should not be using a mix, it should be something realistic that someone might use, which would normally be doing something like titles with "small", "medium" and "large" or "thin", "medium" and "thick".

{{site.requires_5_9v}}

This option is used to specify a list of pre-defined cell border widths for quick access on the `tablecellborderstyle` toolbar button or menu item, in addition to the dialog options. This option accepts any valid CSS border style. [MDN web docs - Border style](https://developer.mozilla.org/en-US/docs/Web/CSS/border-style#values) 

**Type:** `Array`

**Default Value:** 
```
[
  {title: 'Solid', value: 'solid'},
  {title: 'Dotted', value: 'dotted'},
  {title: 'Dashed', value: 'dashed'},
  {title: 'Double', value: 'double'},
  {title: 'Groove', value: 'groove'},
  {title: 'Ridge', value: 'ridge'},
  {title: 'Inset', value: 'inset'},
  {title: 'Outset', value: 'outset'},
  {title: 'None', value: 'none'},
  {title: 'Hidden', value: 'hidden'}
]
```

#### Example: Using `table_border_styles`

```js
tinymce.init({
  selector: 'textarea',  // change this value according to your HTML
  plugins: 'table',
  menubar: 'table',
  toolbar: 'table',
  table_border_styles: [
    {title: 'Solid', value: 'solid'},
    {title: 'Dotted', value: 'dotted'},
    {title: 'Dashed', value: 'dashed'}
  ]
});
```

### `table_cell_background_color_map`

{{site.requires_5_9v}}

> This option is used to specify the default values for the table cell background color picker `tablecellbackgroundcolor`. If no values are defined, the toolbar button and menu item will use the values or default values of the [`color_map` option]({{site.baseurl}}/plugins/opensource/table/#color_map). The option can handle Hex, sRGB and HSL, and ignores alpha values.

Reviewer: ```suggestion
This option is used to specify the default values for the table cell background color picker `tablecellbackgroundcolor`. If no values are defined, the toolbar button and menu item will use the values or default values of the [`color_map` option]({{site.baseurl}}/plugins/opensource/table/#color_map). This option accepts Hex, RGB and HSL color values.
```

The [custom color picker]({{site.baseurl}}/configure/content-appearance/#custom_colors) is not available for the `tablecellbackgroundcolor` toolbar button or menu item.

**Type:** `Array`

**Default Value:** See [`color_map` option]({{site.baseurl}}/plugins/opensource/table/#color_map)

#### Example: Using `table_cell_background_color_map`

```js
tinymce.init({
  selector: 'textarea',  // change this value according to your HTML
  plugins: 'table',
  menubar: 'table',
  toolbar: 'table',
  table_cell_background_color_map: [
    {title: 'Red', value: 'FF0000'},
    {title: 'White', value: 'FFFFFF'},
    {title: 'Yellow', value: 'F1C40F'}
  ]
});
```

### `table_cell_border_color_map`

{{site.requires_5_9v}}

> This option enables you to specify the default values for the table cell border color picker, which can be opened with the `tablecellbordercolor` toolbar button or menu item. If no values are defined, uses the values of the [`color_map` option]({{site.baseurl}}/plugins/opensource/table/#color_map). The option can handle Hex, sRGB and HSL, and ignores alpha values.

Reviewer: ```suggestion
This option enables you to specify the default values for the table cell border color picker, which can be opened with the `tablecellbordercolor` toolbar button or menu item. If no values are defined, uses the values of the [`color_map` option]({{site.baseurl}}/plugins/opensource/table/#color_map). This option accepts Hex, RGB and HSL color values.
```

});
```

### `table_border_widths`

{{site.requires_5_9v}}

> This option is used to specify a list of pre-defined cell border widths for quick access on the `tablecellborderwidth` toolbar button or menu item. This option accepts any valid css numberic value. [MDN web docs - Numeric data types](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Values_and_Units#numeric_data_types)

Reviewer 2: ```suggestion
This option is used to specify a list of pre-defined cell border widths for quick access on the `tablecellborderwidth` toolbar button or menu item. This option accepts any valid CSS numeric value. [MDN web docs - Numeric data types](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Values_and_Units#numeric_data_types)
```

**Type:** `Array`

**Default Value:** 
```
[
  {title: '1px', value: '1px'},
  {title: '2px', value: '2px'},
  {title: '3px', value: '3px'},
  {title: '4px', value: '4px'},
  {title: '5px', value: '5px'}
]
```

#### Example: Using `table_border_widths`

```js
tinymce.init({
  selector: 'textarea',  // change this value according to your HTML
  plugins: 'table',
  menubar: 'table',
  toolbar: 'table',
  table_border_widths: [
    {title: '1 pixel', value: '1px'},
    {title: '2 pixels', value: '2px'},
    {title: '3 pixels', value: '3px'},

>     {title: '5 pixels', value: '5px'}

Reviewer 2: It's probably worth showing the different CSS units you can do here

  ]
});
```

### `table_border_styles`

{{site.requires_5_9v}}

> This option is used to specify a list of pre-defined cell border widths for quick access on the `tablecellborderstyle` toolbar button or menu item, in addition to the dialog options. This option accepts any valid css borders style. [MDN web docs - Border style](https://developer.mozilla.org/en-US/docs/Web/CSS/border-style#values) 

Reviewer 2: ```suggestion
This option is used to specify a list of pre-defined cell border widths for quick access on the `tablecellborderstyle` toolbar button or menu item, in addition to the dialog options. This option accepts any valid CSS border style. [MDN web docs - Border style](https://developer.mozilla.org/en-US/docs/Web/CSS/border-style#values) 
```

**Default Value:** 
```
[
  {title: 'Solid', value: 'solid'},
  {title: 'Dotted', value: 'dotted'},
  {title: 'Dashed', value: 'dashed'},
  {title: 'Double', value: 'double'},
  {title: 'Groove', value: 'groove'},
  {title: 'Ridge', value: 'ridge'},
  {title: 'Inset', value: 'inset'},
  {title: 'Outset', value: 'outset'},
  {title: 'None', value: 'none'},
  {title: 'Hidden', value: 'hidden'}
]
```

#### Example: Using `table_border_styles`

```js
tinymce.init({
  selector: 'textarea',  // change this value according to your HTML
  plugins: 'table',
  menubar: 'table',
  toolbar: 'table',
  table_border_styles: [
    {title: 'Solid', value: 'solid'},
    {title: 'Dotted', value: 'dotted'},
    {title: 'Dashed', value: 'dashed'}
  ]
});
```

### `table_cell_background_color_map`

{{site.requires_5_9v}}

This option is used to specify the default values for the table cell background color picker `tablecellbackgroundcolor`. If no values are defined, the toolbar button and menu item will use the values or default values of the [`color_map` option]({{site.baseurl}}/plugins/opensource/table/#color_map). The option can handle Hex, sRGB and HSL, and ignores alpha values.

> Custom colors can't be enabled.

Reviewer 3: "Custom colors can't be enabled."

what is meant by this?  I assume we can configure any valid hex/sRGB, HSL color to the map and call it anything we like?

Author: Yes, you can configure any valid color. However, when using the colorswatch the user can't select custom colors of their own choice, as they can with fontcolors and background color. This is due to a techdebt we intend to solve later.

Reviewer 2: Up to @tylerkelly13 
```suggestion
Custom colors cannot be enabled.
```

Reviewer 4: ```suggestion
The [custom color picker]({{site.baseurl}}/configure/content-appearance/#custom_colors) is not available for the `tablecellbackgroundcolor` toolbar button or menu item.
```

**Type:** `Array`

**Default Value:** See [`color_map` option]({{site.baseurl}}/plugins/opensource/table/#color_map)

#### Example: Using `table_cell_background_color_map`

```js
tinymce.init({
  selector: 'textarea',  // change this value according to your HTML
  plugins: 'table',
  menubar: 'table',
  toolbar: 'table',
  table_cell_background_color_map: [
    {title: 'Red', value: 'FF0000'},
    {title: 'White', value: 'FFFFFF'},
    {title: 'Yellow', value: 'F1C40F'}
  ]
});
```

### `table_cell_border_color_map`

{{site.requires_5_9v}}

> This option enables you to specify the default values for the table cell background color picker `tablecellbordercolor`. If no values are defined, uses the values of the [`color_map` option]({{site.baseurl}}/plugins/opensource/table/#color_map). The option can handle Hex, sRGB and HSL, and ignores alpha values.

Reviewer 2: This needs a bit of work. The button name just inserted in there doesn't make much sense, so I think if we want to keep it we should move it.

This applies to the background color paragraph above too.

```suggestion
This option enables you to specify the default values for the table cell border color picker, which can be opened with the `tablecellbordercolor` toolbar button or menu item. If no values are defined, uses the values of the [`color_map` option]({{site.baseurl}}/plugins/opensource/table/#color_map). The option can handle Hex, sRGB and HSL, and ignores alpha values.
```

});
```

> ### `table_border_widths`

Author: Any CSS unit, it's basically just put directly in without any need for manipulation.

{{site.requires_5_9v}}

> This option enables you to specify a list of pre-defined cell border withs for quick access by the `tablecellborderwidth` toolbar button or menu item.

Reviewer 4: ```suggestion
This option is used to specify a list of pre-defined cell border widths for quick access on the `tablecellborderwidth` toolbar button or menu item.
```

**Type:** `Array`

**Default Value:** 
```
[
  {title: '1px', value: '1px'},
  {title: '2px', value: '2px'},
  {title: '3px', value: '3px'},
  {title: '4px', value: '4px'},
  {title: '5px', value: '5px'}
]
```

#### Example: Using `table_border_widths`

```js
tinymce.init({
  selector: 'textarea',  // change this value according to your HTML
  plugins: 'table',
  menubar: 'table',
  toolbar: 'table',
  table_border_widths: [
    {title: '1px', value: '1px'},
    {title: '2px', value: '2px'},
    {title: '3px', value: '3px'},
    {title: '4px', value: '4px'},
    {title: '5px', value: '5px'}
  ]
});
```

### `table_border_styles`

{{site.requires_5_9v}}

> This option enables you to specify a list of pre-defined cell border widths for quick access by the `tablecellborderstyle` toolbar button or menu item.

Reviewer: This also impacts the dialog IIRC?

Author: Good catch.

> This option enables you to specify a list of pre-defined cell border widths for quick access by the `tablecellborderstyle` toolbar button or menu item, in addition to the dialog options.

Reviewer 4: ```suggestion
This option is used to specify a list of pre-defined cell border widths for quick access on the `tablecellborderstyle` toolbar button or menu item, in addition to the dialog options.
```

**Type:** `Array`

**Default Value:** 
```
[
  {title: 'Solid', value: 'solid'},
  {title: 'Dotted', value: 'dotted'},
  {title: 'Dashed', value: 'dashed'},
  {title: 'Double', value: 'double'},
  {title: 'Groove', value: 'groove'},
  {title: 'Ridge', value: 'ridge'},
  {title: 'Inset', value: 'inset'},
  {title: 'Outset', value: 'outset'},
  {title: 'None', value: 'none'},
  {title: 'Hidden', value: 'hidden'}
]
```

#### Example: Using `table_border_styles`

```js
tinymce.init({
  selector: 'textarea',  // change this value according to your HTML
  plugins: 'table',
  menubar: 'table',
  toolbar: 'table',
  table_border_styles: [
    {title: 'Solid', value: 'solid'},
    {title: 'Dotted', value: 'dotted'},
    {title: 'Dashed', value: 'dashed'}
  ]
});
```

### `table_cell_background_color_map`

{{site.requires_5_9v}}

> This option enables you to specify the default values for the table cell background color picker `tablecellbackgroundcolor`. If no values are defined, uses the values of `color_map`.

Reviewer 4: ```suggestion
This option is used to specify the default values for the table cell background color picker `tablecellbackgroundcolor`. If no values are defined, the toolbar button and menu item will use the values or default values of the `color_map` option.
```

> Custom colors are not enabled.

Author: No, custom colors are entirely disabled for these, and can't be enabled by the users at all.
For now.
